Tools/clinic/clinic.py

```diff
@@ -927,7 +929,8 @@ def parser_body(prototype, *fields, declarations=''):
 
             add_label = None
             for i, p in enumerate(parameters):
-                parsearg = p.converter.parse_arg(argname_fmt % i, i + 1)
+                displayname = p.displayname()
```

Comment:
Wouldn't be better to generate the displayed name for all kinds of arguments? `"argument '{p.name}'"` for named parameters, `"argument {i + 1}"` for positional-only parameters and just `"argument"` for a single positional-only parameter. This would simplify `_PyArg_BadArgument()`.

Reply:
Thanks, it's better now.